frontend/src/pages/RoadmapGraphPage.tsx

```diff
@@ -13,24 +13,19 @@ import { InfoTooltip } from '../components/InfoTooltip';
 import css from './RoadmapGraphPage.module.scss';
 import { BusinessIcon, WorkRoundIcon } from '../components/RoleIcons';
 import { BlockGraph, BlockView } from '../components/BlockGraph';
+import { RoadmapGraphSidebar } from '../components/RoadmapGraphSidebar';
 import { chosenRoadmapIdSelector } from '../redux/roadmaps/selectors';
-import { Version } from '../redux/roadmaps/types';
+import { VersionComplexityAndValues } from '../redux/roadmaps/types';
 import colors from '../colors.module.scss';
 import { apiV2 } from '../api/api';
 
 const classes = classNames.bind(css);
 
-interface VersionComplexityAndValues extends Version {
-  complexity: number;
-  value: number;
-  totalValue: number;
-  unweightedValue: number;
-  unweightedTotalValue: number;
-}
-
 export const RoadmapGraphPage = () => {
   const { t } = useTranslation();
   const [selectedVersion, setSelectedVersion] = useState(-1);
+  const [sidebarWidth, setSidebarWidth] = useState(0);
+  const [sidebarRef, setSidebarRef] = useState<HTMLDivElement | null>(null);
```

Comment:
useRef does not notify about changes to the value, so it did not go well with the useEffect. [Suggested solution](https://reactjs.org/docs/hooks-faq.html#how-can-i-measure-a-dom-node) with useCallback is not ideal either, since we want to re-measure the width if a different version is selected.